service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -852,6 +855,86 @@ public List<FilterSitesResponse> filterSites(
         .collect(Collectors.toList());
   }
 
+  @Override
+  public List<FilterSiteImageResponse> filterSiteImages(
+      JWTData userData, FilterSiteImageRequest filterSiteImageRequest) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+
+    Condition filterCondition =
+        SITE_IMAGES.APPROVAL_STATUS.eq(String.valueOf(ImageApprovalStatus.SUBMITTED));
```

Comment:
I might be looking at the wrong spot but it seems like you aren't reusing the getUnapprovedImages route. However, I think this is fine since the logic would get pretty messy.

Reply:
This is a good point and I did look at getUnapprovedImages. However, I think the filtering functionality makes it a little bit messy. If you have more to say on what you're thinking with getUnapprovedImages though I can revisit this

Reply:
I think `getUnapprovedImages` was from the very start of building the site images stuff, when we hadn't yet decided to add filtering on anything. I don't think we'll have a reason to use this route anymore unfortunately so we'll probably want to remove it at some point